### PR TITLE
u-u: implement plugin system with `postrun()` functionality

### PR DESCRIPTION
**Pull request by mvo5**
[opening as draft, this needs a bit more work, I think the right way is to add a namedtuple or a dataclass for `class UnattendedUpgradeResult` 

This commit implements a new plugin system. Plugins are python classes that start with `UnattendedUpgradesPlugin` and that are locates in one of the following directories:
```
/etc/unattended-upgrades/plugins/
/usr/share/unattended-upgrades/plugins
```
Plugins are also searched in the `UNATTENDED_UPGRADES_PLUGIN_PATH` environment path and in any of the directories in the apt configration: `Unattended-Upgrade::Dirs::Plugins::`.

The `postrun()` function is called with a single python dictionary as the argument. This dictionary contains the following elements:
* "plugin-api": the API version as string of the form "1.0"
* "hostname": The hostname of the machine that run u-u.
* "success": A boolean that indicates if the run was successful
* "result-str": A string with a human readable (and translated) status message
* "packages-upgraded": A list of packages that got upgraded.
* "packages-kept-back": A list of packages kept back.
* "packages-kept-installed": A list of packages not auto-removed.
* "reboot-required": Indicates a reboot is required.
* "log-dpkg": The full dpkg log output.
* "log-unattended-upgrades": The full unattended-upgrades log.

This new code can be tested with:
```
$ sudo PYTHONPATH=. UNATTENDED_UPGRADES_PLUGIN_PATH=./examples/plugins/ ./unattended-upgrade
```

### CODE REVIEW

**Comment by papamoose** [comment]
After changing some keys in my apprise plugin this still seems to work as expected. I've deployed it where I'm able for testing.

**Comment by mvo5** [comment]
> After changing some keys in my apprise plugin this still seems to work as expected. I've deployed this PR where I'm able for testing.

Thanks for the testing and sorry for these slightly incompatibilities in this final version, once this is merged/released that will (obviously) not change again :)

**Comment by mvo5** [comment]
I tweaked the documentation for the new feature a little bit and I think it's ready now . If someone could double check for clarity/spelling/typos that would be appreciated! I want to merge this soon(ish).

**Comment by papamoose** [comment]
> After changing some keys in my apprise plugin this still seems to work as expected. I've deployed this PR where I'm able for testing.

Looks good so far. All systems are reporting in as they should.

File: README.md
Comment (papamoose) [inline README.md:1]: @mvo5 maybe rephrase slightly like this:

Plugin support is available to integrate with webhooks or other custom tools that need to read u-u run results. Check the [example plugin](https://github.com/mvo5/unattended-upgrades/blob/master/examples/plugins/simple.py) in the git repository [for more details].

Comment (mvo5) [reply README.md:1]: Thank you! That is a great suggestion.
